dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/AbstractTimerTask.java

```diff
@@ -69,19 +69,13 @@ private void reput(Timeout timeout, Long tick) {
 
     @Override
     public void run(Timeout timeout) throws Exception {
-        Collection<Channel> c = channelProvider.getChannels();
-        for (Channel channel : c) {
-            if (channel.isClosed()) {
-                continue;
-            }
-            doTask(channel);
-        }
-        reput(timeout, tick);
+            doTask(channel, timeout);
     }
 
-    protected abstract void doTask(Channel channel);
+    protected abstract void doTask(Channel channel, Timeout timeout);
 
-    interface ChannelProvider {
-        Collection<Channel> getChannels();
+    protected Long getTick() {
```

Comment:
I think we should not allowed to override getTick, if sub class implement this and set value as zero then we might cause issue where we are using tick value for reput , e.g see HeartbeatTimerTask.java doTask method where we are checking whether heart beat value is > getTick and then creating request.

Could you double check this one, I might be wrong.